components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -363,6 +374,26 @@ protected void unwrapResponse(CommonAuthResponseWrapper responseWrapper, String
         }
     }
 
+    private boolean isIdentifierFirstRequest(HttpServletRequest request) {
+
+        String authType = request.getParameter(AUTH_TYPE);
+        return IDF.equals(authType) || request.getParameter(IDENTIFIER_CONSENT) != null;
+    }
+
+    private boolean isIDFAuthenticatorInCurrentStep(AuthenticationContext context) {
+
+        StepConfig stepConfig = context.getSequenceConfig().getStepMap().get(context.getCurrentStep());
+        if (stepConfig != null) {
+            List<AuthenticatorConfig> authenticatorList = stepConfig.getAuthenticatorList();
+            for (AuthenticatorConfig config : authenticatorList) {
```

Comment:
`stepConfig.getAuthenticatorList()` can be null?

Reply:
Authenticatorlist of a step can't be null as we are initializing it here,
https://github.com/wso2/carbon-identity-framework/blob/a1cc113116bb31208945f3338030da43fc4d45f7/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/StepConfig.java#L42